### PR TITLE
Adds GCP gce authentication example

### PR DESCRIPTION
**Pull request by petems**
Adding iam after this is merged 👍

### CODE REVIEW

File: identity/gcp-gce-auth/README.md [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+exec -l $SHELL
+gcloud init
+```
+
+Configure authentication:
+
+```
+gcloud auth login
+gcloud auth application-default login
+```
```
Comment (yhyakuna) [inline identity/gcp-gce-auth/README.md]: I still get an authorization error during `terraform apply`:
```
Error: Error applying plan:

1 error(s) occurred:

* google_project.vault_gcp_demo: 1 error(s) occurred:

* google_project.vault_gcp_demo: Error creating project vault-gcp-demo-68528d20 (vault-gcp-demo): googleapi: Error 403: User is not authorized., forbidden.
```
Are there additional steps that're missing in the instruction?  I found [this](https://github.com/hashicorp/terraform/issues/13109).

File: identity/gcp-gce-auth/README.md [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+gcloud beta billing accounts list
+
+export TF_VAR_org_id=<Organisation ID>
+export TF_VAR_billing_account=<Billing Account ID>
+```
+
+Run terraform:
+
+```
+terraform plan
```
[the remark this answers is not in the document]
Comment (petems) [reply identity/gcp-gce-auth/README.md]: Fixed! 👍

File: identity/gcp-gce-auth/README.md (hunk 1 above)
Comment (petems) [reply identity/gcp-gce-auth/README.md]: @yhyakuna You free to do a screenshare to go through this?

Comment (yhyakuna) [reply identity/gcp-gce-auth/README.md]: The error may be due to my account setting.

File: identity/gcp-gce-auth/README.md [hunk 3]
```diff
@@ -124,6 +124,12 @@ Initialize Vault:
...
 
+Unseal Vault (by supplying at least 3 keys from the above output)
+```
+vault operator unseal
+Unseal Key (will be hidden): <Enter key here>
+```
+
 Export the Vault token from the info from the initialize
 ```
 export VAULT_TOKEN=7a902c8b-a14e-f2af-e007-4482218c586e
```
Comment (yhyakuna) [inline identity/gcp-gce-auth/README.md]: A very minor picky little thing...  The initial root token now looks like `s.KkNJYWF5g0pomcCLEmDdOVCW`.   (I was updating getting started guide, so happened to notice...)

Comment (petems) [reply identity/gcp-gce-auth/README.md]: Good point, I can fix this now 👍

**Pull request merged**